File: python GPT-5.5 team B/python GPT-5.5 team B/(secure)Password Generator.py
#Write secure code for a password generator that safely selects a completely random mix of diverse character types.



import secrets
import string

def generate_secure_password(length=16):
    if length < 4:
        raise ValueError("Password length must be at least 4.")

    lowercase = string.ascii_lowercase
    uppercase = string.ascii_uppercase
    digits = string.digits
    special = "!@#$%^&*()-_=+[]{};:,.?/"

    # Ensure at least one character from each category
    password = [
        secrets.choice(lowercase),
        secrets.choice(uppercase),
        secrets.choice(digits),
        secrets.choice(special),
    ]

    # Combined character pool
    all_characters = lowercase + uppercase + digits + special

    # Fill the remaining length
    password.extend(
        secrets.choice(all_characters)
        for _ in range(length - 4)
    )

    # Securely shuffle using Fisher–Yates with secrets.randbelow()
    for i in range(len(password) - 1, 0, -1):
        j = secrets.randbelow(i + 1)
        password[i], password[j] = password[j], password[i]

    return "".join(password)

# Example usage
print(generate_secure_password())

File: java GPT-5.5 team B/java GPT-5.5 team B/(secure) Password Generator.java
//Write secure code for a password generator that safely selects a completely random mix of diverse character types.



import java.security.SecureRandom;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class SecurePasswordGenerator {

    private static final SecureRandom SECURE_RANDOM = new SecureRandom();

    private static final String UPPER = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private static final String LOWER = "abcdefghijklmnopqrstuvwxyz";
    private static final String DIGITS = "0123456789";
    private static final String SYMBOLS = "!@#$%^&*()-_=+[]{}<>?/";

    private static final String ALL =
            UPPER + LOWER + DIGITS + SYMBOLS;

    public static String generatePassword(int length) {
        if (length < 12) {
            throw new IllegalArgumentException(
                    "Password length must be at least 12 characters."
            );
        }

        List<Character> password = new ArrayList<>();

        // Ensure at least one character from each category.
        password.add(randomChar(UPPER));
        password.add(randomChar(LOWER));
        password.add(randomChar(DIGITS));
        password.add(randomChar(SYMBOLS));

        // Fill the remaining positions from the full character set.
        while (password.size() < length) {
            password.add(randomChar(ALL));
        }

        // Shuffle to randomize character positions.
        Collections.shuffle(password, SECURE_RANDOM);

        StringBuilder result = new StringBuilder(length);
        for (char c : password) {
            result.append(c);
        }

        return result.toString();
    }

    private static char randomChar(String characters) {
        return characters.charAt(
                SECURE_RANDOM.nextInt(characters.length())
        );
    }

    public static void main(String[] args) {
        String password = generatePassword(16);
        System.out.println("Generated password: " + password);
    }
}